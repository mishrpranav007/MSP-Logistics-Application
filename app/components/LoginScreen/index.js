import React from 'react';
import {
  StyleSheet,
  View,
  ImageBackground,
  Image,
  TextInput,
  TouchableOpacity,
  Text
} from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
import logo from 'app/assets/images/imgMspLogo.png';
import deliveryboyimage from 'app/assets/images/illoDeliveryBoy.png';
import homepage from 'app/assets/images/background.png';

const LoginScreen = () => (
  <ImageBackground source={homepage} style={styles.backgroundContainer}>
    <View>
      <Image source={logo} style={styles.logoContainer}></Image>
    </View>
    <View style={styles.deliveryboyimagecontainer}>
      <Image source={deliveryboyimage}></Image>
    </View>
    <View>
      <Icon
        name="ios-person"
        size={20}
        color="rgba(255,255,255,0.7)"
        style={styles.inputIcon}
      />
      <TextInput
        style={styles.input}
        placeholder="Enter your email address"
        placeholderTextColor="rgb(255,255,255)"
      />
    </View>
    <View>
      <Icon
        name="ios-lock"
        size={20}
        color="rgba(255,255,255,0.7)"
        style={styles.inputIcon}
      />
      <TextInput
        style={styles.input}
        placeholder="Enter password"
        placeholderTextColor="rgb(255,255,255)"
      />
    </View>

    <TouchableOpacity style={styles.btnLogin}>
      <Text style={styles.text}>Login</Text>
    </TouchableOpacity>
  </ImageBackground>
);
const styles = StyleSheet.create({
  backgroundContainer: {
    flex: 1,
    justifyContent: 'center',
    width: null,
    height: null,
    alignItems: 'center',
    flexDirection: 'column'
  },
  logoContainer: {
    width: 192,
    height: 51,
    marginTop: 67,
    marginLeft: 92,
    marginRight: 88,
    justifyContent: 'flex-start',
    alignItems: 'center',
    marginBottom: 8
  },
  deliveryboyimagecontainer: {
    width: 249,
    height: 256,
    marginLeft: 73,
    marginTop: 13,
    marginBottom: 38,
    justifyContent: 'center',
    alignItems: 'flex-end'
  },

  input: {
    width: 270,
    backgroundColor: 'transparent',
    borderBottomWidth: 1,
    borderBottomColor: 'white',
    marginTop: 20,
    position: 'relative',
    paddingLeft: 26,
    fontSize: 14
  },
  btnLogin: {
    width: 310,
    height: 48,
    justifyContent: 'center',
    backgroundColor: 'rgb(36,106,116)',
    marginTop: 48,
    borderRadius: 11,
    marginBottom: 55
  },
  text: {
    color: 'rgb(255,255,255)',
    fontSize: 16,
    textAlign: 'center'
  },
  inputIcon: {
    position: 'absolute',
    left: 0,
    bottom: 0,
    marginBottom: 0,
    justifyContent: 'space-between'
  }
});
export default LoginScreen;
