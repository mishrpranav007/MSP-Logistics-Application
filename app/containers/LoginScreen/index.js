import React, { Component } from 'react';
import {
  StyleSheet,
  View,
  ImageBackground,
  KeyboardAvoidingView,
  Image,
  TextInput,
  TouchableOpacity,
  Text,
  Platform
} from 'react-native';
import { Formik } from 'formik';
import * as yup from 'yup';
import { injectIntl } from 'react-intl';
import Icon from 'react-native-vector-icons/MaterialIcons';
import logo from 'app/assets/images/imgMspLogo.png';
import deliveryboyimage from 'app/assets/images/illoDeliveryBoy.png';
import homepage from 'app/assets/images/loginbg.jpg';
import DismissableKeyboard from 'app/components/DismissKeyboard/';
import PropTypes from 'prop-types';
import Style from '../../style/index';

class LoginScreen extends Component {
  render() {
    return (
      <ImageBackground source={homepage} style={styles.backgroundContainer}>
        <DismissableKeyboard>
          <KeyboardAvoidingView
            style={styles.container}
            behavior={Platform.Os === 'ios' ? 'padding' : 'position'}
            enabled
          >
            <View>
              <Image source={logo} style={styles.logoContainer}></Image>
            </View>

            <View style={styles.deliveryboyImageContainer}>
              <Image source={deliveryboyimage}></Image>
            </View>
            <Formik
              initialValues={{ email: '', password: '' }}
              // eslint-disable-next-line no-alert
              onSubmit={values => alert(JSON.stringify(values))}
              validationSchema={yup.object().shape({
                email: yup
                  .string()
                  .email()
                  .required(),

                password: yup
                  .string()
                  .min(6)
                  .required()
              })}
            >
              {({
                values,
                handleChange,
                errors,
                setFieldTouched,
                touched,
                handleSubmit
              }) => (
                <View>
                  <View>
                    <Icon
                      name="person-outline"
                      size={Style.em(1.375)}
                      color={Style.ICON_COLOR}
                      style={styles.inputIcon}
                    />
                    <TextInput
                      testID="log-in-screen-text-input"
                      value={values.email}
                      style={styles.input}
                      placeholder={this.props.intl.formatMessage({
                        id: 'placeholder_email'
                      })}
                      onBlur={() => setFieldTouched('email')}
                      returnKeyType="next"
                      onSubmitEditing={() => {
                        this.secondTextInput.focus();
                      }}
                      placeholderTextColor={Style.WHITE_COLOR}
                      onChangeText={handleChange('email')}
                    />
                  </View>
                  <View>
                    {touched.email && errors.email && (
                      <Text
                        style={{
                          position: 'relative',
                          marginLeft: Style.em(4.09),
                          color: 'red'
                        }}
                      >
                        {errors.email}
                      </Text>
                    )}
                  </View>
                  <View behavior="position">
                    <Icon
                      name="lock-outline"
                      size={Style.em(1.375)}
                      color={Style.ICON_COLOR}
                      style={styles.inputIcon}
                    />
                    <TextInput
                      testID="log-in-screen-text-input"
                      value={values.password}
                      ref={input => {
                        this.secondTextInput = input;
                      }}
                      style={styles.input}
                      placeholder={this.props.intl.formatMessage({
                        id: 'placeholder_password'
                      })}
                      returnKeyType="go"
                      placeholderTextColor={Style.WHITE_COLOR}
                      onChangeText={handleChange('password')}
                      onBlur={() => setFieldTouched('password')}
                    />
                  </View>
                  <View>
                    {touched.password && errors.password && (
                      <Text
                        style={{
                          color: 'red',
                          position: 'relative',
                          marginLeft: Style.em(4.09)
                        }}
                      >
                        {errors.password}
                      </Text>
                    )}
                  </View>
                  <TouchableOpacity
                    style={styles.btnLogin}
                    onPress={handleSubmit}
                  >
                    <Text style={styles.text}>Login</Text>
                  </TouchableOpacity>
                </View>
              )}
            </Formik>
          </KeyboardAvoidingView>
        </DismissableKeyboard>
      </ImageBackground>
    );
  }
}
const styles = StyleSheet.create({
  backgroundContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'column'
  },
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  },
  logoContainer: {
    width: Style.em(12.375),
    height: Style.em(3.313),
    marginTop: Platform.OS === 'ios' ? Style.em(3.125) : Style.em(5.15),
    marginLeft: Style.em(5.75),
    marginRight: Style.em(5.5),
    justifyContent: 'flex-start',
    alignItems: 'center',
    marginBottom: Style.em(1)
  },
  deliveryboyImageContainer: {
    width: Style.em(15.563),
    height: Style.em(16),
    marginLeft: Style.em(7.68),
    marginTop: Style.em(0.813),
    marginBottom: Platform.OS === 'ios' ? Style.em(2.375) : Style.em(1.5),
    justifyContent: 'center',
    alignItems: 'flex-end'
  },

  input: {
    width: Style.em(17),
    backgroundColor: 'transparent',
    borderBottomWidth: 1,
    borderBottomColor: 'white',
    marginTop: Platform.OS === 'ios' ? Style.em(1.813) : Style.em(0.92),
    position: 'relative',
    marginLeft: Style.em(3.75),
    paddingLeft: Style.em(1.625),
    paddingBottom: Platform.OS === 'android' ? Style.em(0) : null,
    zIndex: 30,
    opacity: 0.94,
    color: Style.WHITE_COLOR,
    fontSize: Platform.OS === 'ios' ? Style.em(0.875) : Style.em(1)
  },
  btnLogin: {
    width: Style.em(19.375),
    height: Style.em(3),
    justifyContent: 'center',
    backgroundColor: Style.PRIMARY_COLOR,
    marginTop: Platform.OS === 'ios' ? Style.em(3.063) : Style.em(2),
    marginLeft: Style.em(2.313),
    borderRadius: Style.em(0.688),
    marginBottom: Platform.OS === 'ios' ? Style.em(3.25) : Style.em(5),
    borderTopColor: '1px solid rgba(19,43,46,0.83)',
    zIndex: 100
  },
  text: {
    color: Style.WHITE_COLOR,
    fontSize: Style.em(1.125),
    textAlign: 'center'
  },
  inputIcon: {
    position: 'absolute',
    left: 0,
    bottom: 0,
    marginLeft: Style.em(3.438),
    justifyContent: 'space-between',
    opacity: 0.94
  }
});

LoginScreen.propTypes = {
  intl: PropTypes.object
};
export default injectIntl(LoginScreen);
