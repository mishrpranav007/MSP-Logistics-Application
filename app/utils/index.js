import { Platform, Dimensions } from 'react-native';

export const isAndroid = () => Platform.OS !== 'ios';
export const getScreenWidth = () => Dimensions.get('window').width;
export const getScreenHeight = () => Dimensions.get('window').height;

export const getPlatformString = () => (isAndroid() ? 'android' : 'ios');
export const getFontFamily = font =>
  isAndroid() ? `Futura-${font}` : `FuturaStd-${font}`;

export const isScreenSmall = () => getScreenHeight() <= 550;
