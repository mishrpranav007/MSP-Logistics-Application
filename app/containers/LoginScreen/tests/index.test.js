import React from 'react';
import { fireEvent, act } from '@testing-library/react-native';
import { renderWithIntl } from 'app/utils/testUtils';
import { Formik } from 'formik';
import TextInput from 'react-native';
import LoginScreen from '../index';

describe('<LoginScreen />', () => {
  const submitSpy = jest.fn();
  beforeEach(() => {
    submitSpy();
  });
  it('should render and match the snapshot', async () => {
    const { baseElement } = await renderWithIntl(<LoginScreen />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain the 2 textinput component inside the Loginscreen', async () => {
    const { getAllByTestId } = await renderWithIntl(<LoginScreen />);
    expect(getAllByTestId('log-in-screen-text-input').length).toBe(2);
  });
  it('should contain a login button component inside the Loginscreen', async () => {
    const { getAllByTestId } = await renderWithIntl(<LoginScreen />);
    expect(getAllByTestId('log-in-screen-button').length).toBe(1);
  });
  it('should have correct email value passed to it', async () => {
    const emailText = 'abc123@gmail.com';
    const { getAllByTestId } = renderWithIntl(
      <LoginScreen>
        <Formik initialValues={{ email: '', password: '' }} />
      </LoginScreen>
    );
    act(async () => {
      fireEvent.changeText(
        getAllByTestId('log-in-screen-text-input')[0],
        emailText
      );
    });
    expect(getAllByTestId('log-in-screen-text-input')[0].props.value).toEqual(
      emailText
    );
    expect(
      getAllByTestId('log-in-screen-text-input')[0].props.value
    ).not.toEqual(null);
  });
  it('should have correct password value passed to it', async () => {
    const passwordText = 'qwerty123';
    const { getAllByTestId } = renderWithIntl(
      <LoginScreen>
        <Formik initialValues={{ email: '', password: '' }} />
      </LoginScreen>
    );
    act(async () => {
      fireEvent.changeText(
        getAllByTestId('log-in-screen-text-input')[1],
        passwordText
      );
    });
    expect(getAllByTestId('log-in-screen-text-input')[1].props.value).toEqual(
      passwordText
    );
    expect(
      getAllByTestId('log-in-screen-text-input')[1].props.value
    ).not.toEqual(null);
  });
  it('should check whether setFieldTouched method is getting called when onBlur even is triggered with email input', async () => {
    const { getAllByTestId } = renderWithIntl(
      <LoginScreen>
        <TextInput
          testID="log-in-screen-text-input"
          setFieldTouched={submitSpy}
        />
      </LoginScreen>
    );
    act(async () => {
      fireEvent.blur(getAllByTestId('log-in-screen-text-input')[0]);
    });
    expect(submitSpy).toHaveBeenCalled();
  });
  it('should check whether setFieldTouched method is getting called when onBlur even is triggered with password input', async () => {
    const { getAllByTestId } = renderWithIntl(
      <LoginScreen>
        <TextInput
          testID="log-in-screen-text-input"
          setFieldTouched={submitSpy}
        />
      </LoginScreen>
    );
    act(async () => {
      fireEvent.blur(getAllByTestId('log-in-screen-text-input')[1]);
    });
    expect(submitSpy).toHaveBeenCalled();
  });
  it('should check whether handleChange method is getting called when email input is passed', async () => {
    const emailText = 'johndoe12@gmail.com';
    const { getAllByTestId } = renderWithIntl(
      <LoginScreen>
        <TextInput testID="log-in-screen-text-input" handleChange={submitSpy} />
      </LoginScreen>
    );
    act(async () => {
      fireEvent.changeText(
        getAllByTestId('log-in-screen-text-input')[0].props.value,
        emailText
      );
    });
    expect(submitSpy).toHaveBeenCalled();
  });
  it('should check whether handleChange method is getting called when password input is passed', async () => {
    const passwordText = 'abcdef123';
    const { getAllByTestId } = renderWithIntl(
      <LoginScreen>
        <TextInput testID="log-in-screen-text-input" handleChange={submitSpy} />
      </LoginScreen>
    );
    act(async () => {
      fireEvent.changeText(
        getAllByTestId('log-in-screen-text-input')[1].props.value,
        passwordText
      );
    });
    expect(submitSpy).toHaveBeenCalled();
  });
});
