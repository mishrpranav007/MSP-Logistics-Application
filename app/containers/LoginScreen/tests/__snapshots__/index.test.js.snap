// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginScreen /> should render and match the snapshot 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "opacity": 0.94,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../app/assets/images/loginbg.jpg",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": undefined,
              "width": undefined,
            },
            undefined,
          ]
        }
      />
      <TouchableWithoutFeedback>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "bottom": 0,
              }
            }
          >
            <View>
              <View>
                <Image
                  source={
                    Object {
                      "testUri": "../../../app/assets/images/imgMspLogo.png",
                    }
                  }
                  style={
                    Object {
                      "alignItems": "center",
                      "justifyContent": "flex-start",
                      "marginBottom": 25.142857142857142,
                      "marginLeft": 143.31428571428572,
                      "marginRight": 138.28571428571428,
                      "marginTop": 78.69714285714285,
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "alignItems": "flex-end",
                    "height": 402.2857142857143,
                    "justifyContent": "center",
                    "marginBottom": 59.839999999999996,
                    "marginLeft": 176,
                    "marginTop": 20.365714285714287,
                    "width": 391.22285714285715,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "testUri": "../../../app/assets/images/illoDeliveryBoy.png",
                    }
                  }
                />
              </View>
            </View>
            <View>
              <View>
                <Image
                  source={
                    Object {
                      "testUri": "../../../app/assets/images/username.png",
                    }
                  }
                  style={
                    Object {
                      "borderColor": "rgb(255,255,255)",
                      "bottom": 7.794285714285714,
                      "height": 27.154285714285717,
                      "justifyContent": "center",
                      "left": 0,
                      "marginLeft": 75.42857142857143,
                      "opacity": 0.94,
                      "position": "absolute",
                      "right": 15.84,
                      "width": 22.880000000000003,
                    }
                  }
                />
                <TextInput
                  allowFontScaling={true}
                  placeholder="Enter your email address"
                  placeholderTextColor="rgb(255,255,255)"
                  rejectResponderTermination={true}
                  returnKeyType="next"
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "borderBottomColor": "rgb(255,255,255)",
                      "borderBottomWidth": 1,
                      "color": "rgb(255,255,255)",
                      "fontSize": 21.874285714285715,
                      "marginLeft": 75.42857142857143,
                      "marginTop": 45.50857142857143,
                      "opacity": 0.94,
                      "paddingBottom": null,
                      "paddingLeft": 40.73142857142857,
                      "position": "relative",
                      "width": 427.42857142857144,
                      "zIndex": 30,
                    }
                  }
                  testID="log-in-screen-text-input"
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
              <View />
              <View
                behavior="position"
              >
                <Image
                  source={
                    Object {
                      "testUri": "../../../app/assets/images/password.png",
                    }
                  }
                  style={
                    Object {
                      "borderColor": "rgb(255,255,255)",
                      "bottom": 7.794285714285714,
                      "height": 27.154285714285717,
                      "justifyContent": "center",
                      "left": 0,
                      "marginLeft": 75.42857142857143,
                      "opacity": 0.94,
                      "position": "absolute",
                      "right": 15.84,
                      "width": 22.880000000000003,
                    }
                  }
                />
                <TextInput
                  allowFontScaling={true}
                  placeholder="Enter password"
                  placeholderTextColor="rgb(255,255,255)"
                  rejectResponderTermination={true}
                  returnKeyType="go"
                  secureTextEntry={true}
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "borderBottomColor": "rgb(255,255,255)",
                      "borderBottomWidth": 1,
                      "color": "rgb(255,255,255)",
                      "fontSize": 21.874285714285715,
                      "marginLeft": 75.42857142857143,
                      "marginTop": 45.50857142857143,
                      "opacity": 0.94,
                      "paddingBottom": null,
                      "paddingLeft": 40.73142857142857,
                      "position": "relative",
                      "width": 427.42857142857144,
                      "zIndex": 30,
                    }
                  }
                  testID="log-in-screen-text-input"
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
              <View />
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "rgb(36,106,116)",
                    "borderRadius": 17.348571428571425,
                    "borderTopColor": "rgba(19,43,46,0.83)",
                    "height": 75.42857142857143,
                    "justifyContent": "center",
                    "marginBottom": 81.71428571428572,
                    "marginLeft": 45.25714285714285,
                    "marginTop": 76.93714285714286,
                    "width": 487.77142857142854,
                    "zIndex": 100,
                  }
                }
              >
                <TouchableOpacity
                  activeOpacity={0.2}
                >
                  <Text
                    style={
                      Object {
                        "color": "rgb(255,255,255)",
                        "fontSize": 28.160000000000004,
                        "textAlign": "center",
                      }
                    }
                  >
                    Login
                  </Text>
                </TouchableOpacity>
              </View>
            </View>
          </View>
        </View>
      </TouchableWithoutFeedback>
    </View>
  </View>
</View>
`;
